src/state/sagas/datasets/UpdateDataset/UpdateDataset.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import { takeEvery, call, put } from 'redux-saga/effects';
+import { t } from 'i18next';
+import { Api } from '../../../../services/config/Api';
+import { dispatchSetApplicationErrorMessage } from '../../../dispatchers/app/ApplicationDispatcher';
+import { DATASET_ACTIONS_KEY } from '../../../commons/DatasetConstants';
+
+export function* updateDataset(action) {
+  const organizationId = action.organizationId;
+  const datasetId = action.datasetId;
+  const datasetData = action.datasetData;
+  const selectedDatasetIndex = action.selectedDatasetIndex;
+
+  try {
+    yield call(Api.Datasets.updateDataset, organizationId, datasetId, datasetData);
+    yield put({
+      type: DATASET_ACTIONS_KEY.UPDATE_DATASET,
+      datasetId,
+      datasetData,
+      selectedDatasetIndex,
```

Comment:
I think just `datasetIndex` can be enough for this action because it can be a bit confusing for other cases :slightly_smiling_face: